### PR TITLE
guard against double initialization

### LINKED ISSUE #43: LogManager.initServiceExtensions should be allow multiple calls
Calling `LogManager.initServiceExtensions` more than once should only cause one registration - client code won't know if it's already been called, and may call it itself.

### PR DESCRIPTION
**Pull request by pq**
fixes #43 

/cc @devoncarew

### CODE REVIEW

File: lib/src/log_manager.dart [hunk 1]
```diff
@@ -65,6 +65,8 @@ class LoggingException extends Error implements Exception {
 /// * To enable or disable a logging channel, use [enableLogging].
 /// * To query channel enablement, use [shouldLog].
 class LogManager {
+  // Track initialization state to ensure no double VM service registrations.
+  static bool _initialized;
```
Comment (devoncarew) [inline lib/src/log_manager.dart]: I'd init this to false...

File: lib/src/log_manager.dart [hunk 2]
```diff
@@ -107,6 +109,11 @@ class LogManager {
   /// Note: ideally this will be replaced w/ inline registrations within the
   /// flutter foundation binding (see e.g., https://github.com/flutter/flutter/pull/21505).
   void initServiceExtensions() {
+    // Avoid double initialization.
+    if (_initialized == true) {
```
Comment (devoncarew) [inline lib/src/log_manager.dart]: ...and then just check `if (_initialized)` here.

File: lib/src/log_manager.dart (hunk 1 above)
Comment (pq) [reply lib/src/log_manager.dart]: That's exactly how I started!

File: lib/src/log_manager.dart (hunk 2 above)
Comment (pq) [reply lib/src/log_manager.dart]: ... but this check was failing.  Chalk it up to another nuance in Dart initialization rules but I was hitting an NPE here as `_initialized` was being seen as `null`.  (FWIW Janice was equally flummoxed!)

**Pull request merged**